py-polars/polars/plugins.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from polars._utils.parse_expr_input import parse_as_list_of_expressions
+from polars._utils.udfs import get_dynamic_lib_location
+from polars._utils.unstable import unstable
+from polars._utils.wrap import wrap_expr
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from polars import Expr
+    from polars.type_aliases import IntoExpr
+
+__all__ = ["register_plugin"]
+
+
+@unstable()
+def register_plugin(
+    *inputs: IntoExpr | list[IntoExpr],
+    plugin_location: str | Path,
```

Comment:
I don't think we should accept the `__init__.py` file as an input here. It's convenient for the user but makes no sense from an API standpoint.

This should accept either the exact location of the dynamic library file, or the folder containing it. The util should be updated to reflect this.